peekaboo/sample.py

```diff
@@ -264,29 +267,59 @@ def known(self):
 
     @property
     def file_extension(self):
+        if self.has_attr('file_extension'):
+            return self.get_attr('file_extension')
+
+        # try to find a file name containing an extension. Using
+        # self.__filename will almost never yield anything useful because
+        # amavis intentionally hands us files named only p001, p002 and so on.
+        # But we still try it in case there's no declared name.
```

Comment:
As a side note:
During specification we defined the usage with amavis but also that it should be possible to scan arbitrary files and directories readable by the "peekaboo" user. We already lost this when we changed the "dump_info".

Reply:
I had a feeling there might be something like this and the concept isn't truly dead: Samples can still be created with empty metainfo and providing meta info as json instead of ini-style shouldn't be that hard. Where can I find information on how to scan arbitrary files from the command line?

Reply:
Have a look at `scan_file.py`
https://github.com/scVENUS/PeekabooAV/blob/master/bin/scan_file.py

Reply:
Will do in #46 and come up with a separate PR.